tracing-journald/src/socket.rs

```diff
@@ -0,0 +1,65 @@
+//! socket helpers.
+
+use std::alloc::{alloc_zeroed, dealloc, Layout};
+use std::io::{Error, Result};
+use std::mem::{align_of, size_of, MaybeUninit};
+use std::os::unix::net::UnixDatagram;
+use std::os::unix::prelude::{AsRawFd, RawFd};
+use std::ptr;
+
+use libc::*;
+
+pub fn send_one_fd<F: AsRawFd>(socket: &UnixDatagram, fd: F) -> Result<usize> {
```

Comment:
Do we really need to be generic over multiple possible fd carriers here? This is single-purpose code.

Reply:
Habit, and then I figured it wouldn't do no harm, but you're right, it's simpler without the generic type parameter.  I've removed it.